litmus-portal/graphql-server/utils/misc.go

```diff
@@ -86,8 +88,30 @@ func ManifestParser(cluster dbSchemaCluster.Cluster, rootPath string, subscriber
 		return nil, err
 	}
 
-	for _, fileName := range list {
+	var nodeselector string
+	if cluster.NodeSelector != nil {
+		selector := strings.Split(*cluster.NodeSelector, ",")
+		selectorList := make(map[string]string)
+		for _, el := range selector {
+			kv := strings.Split(el, "=")
+			selectorList[kv[0]] = kv[1]
```

Comment:
This part is kind of repetitive, this is also done in the cluster reg function to verify the data. It would be better to parse it directly in that cluster reg function and store the yaml in db and reuse it directly here.

Reply:
currently, we are not storing yaml in the db.

Reply:
Right now changing the main yaml generation logic before the release is risky

Reply:
I created an issue- https://github.com/litmuschaos/litmus/issues/3003

It's a good feature to have.